### PR TITLE
Added solution for HW_2.2

### PR DESCRIPTION
**Pull request by lord-dwarf**

### CODE REVIEW

File: src/main/java/core/basesyntax/MatrixOperation.java [hunk 1]
```diff
@@ -6,18 +6,38 @@
...
+            counterDiagonal[len - i] = matrix[len - i][i];
+        }
+        return counterDiagonal;
     }
 
     public int getMin(int[][] matrix) {
-        return 0;
+        int min = Integer.MAX_VALUE;
+        for (int row = 0; row < matrix.length; row++) {
+            for (int col = 0; col < matrix.length; col++) {
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/MatrixOperation.java]: names such as i and j would enough here

File: src/main/java/core/basesyntax/MatrixOperation.java [hunk 2]
```diff
@@ -6,18 +6,38 @@
...
+        int[] counterDiagonal = new int[matrix.length];
+        for (int len = matrix.length - 1, i = len; i >= 0; i--) {
+            counterDiagonal[len - i] = matrix[len - i][i];
+        }
+        return counterDiagonal;
     }
 
     public int getMin(int[][] matrix) {
-        return 0;
+        int min = Integer.MAX_VALUE;
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/MatrixOperation.java]: try to use without Integer.MAX_VALUE

File: src/main/java/core/basesyntax/MatrixOperation.java [hunk 3]
```diff
@@ -6,18 +6,38 @@
...
+            for (int col = 0; col < matrix.length; col++) {
+                min = matrix[row][col] < min ? matrix[row][col] : min;
+            }
+        }
+        return min;
     }
 
     public int getMax(int[][] matrix) {
-        return 0;
+        int max = Integer.MIN_VALUE;
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/MatrixOperation.java]: same

File: src/main/java/core/basesyntax/MatrixOperation.java (hunk 1 above)
Comment (lord-dwarf) [reply src/main/java/core/basesyntax/MatrixOperation.java]: tried to add readability. done

File: src/main/java/core/basesyntax/MatrixOperation.java (hunk 2 above)
Comment (lord-dwarf) [reply src/main/java/core/basesyntax/MatrixOperation.java]: done

File: src/main/java/core/basesyntax/MatrixOperation.java (hunk 3 above)
Comment (lord-dwarf) [reply src/main/java/core/basesyntax/MatrixOperation.java]: done

File: src/main/java/core/basesyntax/MatrixOperation.java [hunk 4]
```diff
@@ -6,18 +6,42 @@
  */
 public class MatrixOperation {
     public int[] getDiagonal(int[][] matrix) {
-        return null;
+        int[] principalDiagonal = new int[matrix.length];
+        for (int i = 0; i < matrix.length; i++) {
+            principalDiagonal[i] = (matrix[i][i]);
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/MatrixOperation.java]: do we need these brackets?

File: src/main/java/core/basesyntax/MatrixOperation.java [hunk 5]
```diff
@@ -6,18 +6,42 @@
...
+        }
+        return principalDiagonal;
     }
 
     public int[] getCounterDiagonal(int[][] matrix) {
-        return null;
+        int[] counterDiagonal = new int[matrix.length];
+        for (int len = matrix.length - 1, i = len; i >= 0; i--) {
+            counterDiagonal[len - i] = matrix[len - i][i];
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/MatrixOperation.java]: Try to do that with going through matrix from 0

**Review by dmytro-hryhoruk — APPROVED** [review]

**Pull request closed without merging**